src/coreclr/jit/codegenxarch.cpp

```diff
@@ -504,7 +504,9 @@ void CodeGen::genSetRegToConst(regNumber targetReg, var_types targetType, GenTre
 
             if (vecCon->IsZero())
             {
-                if ((attr != EA_32BYTE) || compiler->compOpportunisticallyDependsOn(InstructionSet_AVX))
+                if (((attr != EA_32BYTE) && (attr != EA_64BYTE)) ||
+                    ((attr == EA_32BYTE) && compiler->compOpportunisticallyDependsOn(InstructionSet_AVX)) ||
+                    ((attr == EA_64BYTE) && compiler->compOpportunisticallyDependsOn(InstructionSet_AVX512F)))
```

Comment:
We could restructure this like:
```suggestion
                bool isSupported;
                
                if (attr == EA_32BYTE)
                {
                    isSupported = compiler->compOpportunisticallyDependsOn(InstructionSet_AVX);
                }
                else if (attr == EA_64BYTE)
                {
                    isSupported = compiler->compOpportunisticallyDependsOn(InstructionSet_AVX512F);
                }
                else
                {
                    assert((attr == EA_8BYTE) || (attr == EA_16BYTE));
                    isSupported = true;
                }

                if (isSupported)
```

I think it's a bit more readable and allows us to assert that the default path is one of the expected sizes.

Reply:
Done